sdk/python/kfp_tekton/compiler/_tekton_handler.py

```diff
@@ -352,6 +363,15 @@ def _handle_tekton_custom_task(custom_task: dict, workflow: dict, recursive_task
             if task['name'].replace(task_name_prefix, "") not in task_list_trimmed:
                 new_tasks.append(task)
     workflow['spec']['pipelineSpec']['tasks'] = new_tasks
+    # clean up any redundant parameters
+    for i, task in enumerate(workflow['spec']['pipelineSpec']['tasks']):
+        searched_params = []
+        if task.get('params'):
+            for j, workflow_param in enumerate(task['params']):
+                if workflow_param['name'] in searched_params:
+                    workflow['spec']['pipelineSpec']['tasks'][i]['params'].pop(j)
```

Comment:
This line pops from an array that is being iterated. Would be better to either store the indexes to remove or storing the new array to assign in one go (and not appending to it in such a case).

Reply:
sure, I rewrote this into a new array instead of popping the items.